pandas/core/internals/array_manager.py

```diff
@@ -330,7 +356,7 @@ def apply_with_block(self: T, f, align_keys=None, **kwargs) -> T:
             if hasattr(arr, "tz") and arr.tz is None:  # type: ignore[union-attr]
                 # DatetimeArray needs to be converted to ndarray for DatetimeBlock
                 arr = arr._data  # type: ignore[union-attr]
-            elif arr.dtype.kind == "m":
+            elif arr.dtype.kind == "m" and not isinstance(arr, np.ndarray):
```

Comment:
could combine this with previous check as

```
if arr.dtype.kind in ["m", "M"] and not isinstance(arr, np.ndarray):
    arr = arr._data
```

Reply:
That would be nice, but the problem is that we still need to keep DatetimeArray intact for DatetimeTZBlock. So we would still need the `if hasattr(arr, "tz") and arr.tz is None` check as well, in which case it doesn't necessarily become more readable to combine both checks.

Reply:
instead of `and getattr(arr, "tz", None) is None` how about `isinstance(arr.dtype, np.dtype)`.  either way works i guess

Reply:
That still gives the same length of the `if` check as in my diff example above, which I don't find an improvement in readability

Reply:
yah the only possible difference is for mypy